hydroflow_plus/src/deploy/mod.rs

```diff
@@ -208,6 +208,18 @@ pub trait ProcessSpec<'a, D: LocalDeploy<'a> + ?Sized> {
     fn build(self, id: usize, name_hint: &str) -> D::Process;
 }
 
+pub trait IntoProcessSpec<'a, D: LocalDeploy<'a> + ?Sized> {
+    type ProcessSpec: ProcessSpec<'a, D>;
+    fn into_process_spec(self) -> Self::ProcessSpec;
+}
```

Comment:
I guess `Ips: IntoProcessSpec<'a, D>` is about the same as `Ips: Into<Ps>, Ps: ProcessSpec<'a, D>`

Reply:
Yes, this is the idea. Although actually doing the `Into<...>` approach runs into type inference failures because there are too many generics involved and Rust gets confused.